core/app/models/spree/shipment.rb

```diff
@@ -130,10 +130,17 @@ def final_price_with_items
       item_cost + final_price
     end
 
+    # Decrement the stock counts for all pending inventory units in this
+    # shipment and mark.
+    # Any previous non-pending inventory units are skipped as their stock had
+    # already been allocated.
     def finalize!
-      return if order.unreturned_exchange?
-      InventoryUnit.finalize_units!(inventory_units)
-      manifest.each { |item| manifest_unstock(item) }
+      transaction do
```

Comment:
:+1: here.

I'm assuming this is to make sure there's only one `begin` and `commit` for he transaction or was this for locking?